test/e2e/healthcheck_test.go

```diff
@@ -0,0 +1,192 @@
+//go:build e2e
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/strings/slices"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kuadrant/dns-operator/api/v1alpha1"
+	"github.com/kuadrant/dns-operator/test/e2e/helpers"
+)
+
+// Test Cases covering multiple DNSRecords updating a set of records in a zone
+var _ = Describe("Health Check Test", func() {
+	Context("DNS Provider health checks", func() {
+		It("creates health checks for a health check spec", func(ctx SpecContext) {
+			probeHost := "app." + testZoneDomainName
+			healthChecksSupported := false
+			if slices.Contains(supportedHealthCheckProviders, strings.ToLower(testDNSProvider)) {
+				healthChecksSupported = true
+			}
+
+			geoCode := "EU"
+			if testDNSProvider == "gcp" {
+				geoCode = "europe-west1"
+			}
+			Expect(os.Setenv("TEST_DNS_GEO_CODE", geoCode)).To(BeNil())
```

Comment:
Hmm, we need to think about what the consequences of using  `os.Setenv` are. Generally, in tests, we have found it to not be a great idea since it's global to all test specs (so can't run in parallel), and you are setting these values on the system running it so you need to make sure you unset it, or put back any old value. The golang testing framework had some helper for this from what i remember.

Reply:
the best I can find is [this](https://www.yellowduck.be/posts/using-environment-variables-in-go-tests), but it seems that ginkgo doesn't expose the `testing.T`. So I will do this via a `BeforeEach/AfterEach` set /unset, if that seems suitable?